docs/rules/prefer-presence-queries.md

```diff
@@ -0,0 +1,63 @@
+# Enforce specific queries when checking appearance or disappearance of elements (prefer-presence-queries)
+
+The (DOM) Testing Library allows to query DOM elements using different types of queries such as `getBy*` and `queryBy*`. Using `getBy*` throws an error in case the element is not found, while `queryBy` returns null instead of throwing. These differences are useful in some situations:
+
+- using `getBy*` queries when asserting if element is present, so if the element is not found the error thrown will offer better info than asserting with other queries which not throw.
+- using `queryBy*` queries when asserting if element is not present, so the test doesn't fail immediately when the element it's not found and the assertion can be executed properly.
+
+## Rule details
+
+This rule fires whenever:
+
+- `queryBy*` is used to assert element **is** present with `.toBeInTheDocument()`, `toBeTruthy()` or `.toBeDefined()` matchers or negated matchers from following case.
+- `getBy*` is used to assert element **is not** present with `.toBeNull()` or `.toBeFalsy()` matchers or negated matchers from previous case.
+
+Examples of **incorrect** code for this rule:
+
+```js
+test('some test', () => {
+  render(<App />);
+
+  // check element is present with `queryBy*`
+  expect(screen.queryByText('button')).toBeInTheDocument();
+  expect(screen.queryByText('button')).toBeTruthy();
+  expect(screen.queryByText('button')).toBeNull();
+  expect(screen.queryByText('button')).not.toBeNull();
+  expect(screen.queryByText('button')).not.toBeFalsy();
```

Comment:
what about the following:

```jsx
const button = screen.queryByText("button");
expect(button).toBeInTheDocument()

// or this one

expect(screen.queryAllByText("button")[1]).toBeInTheDocument()
```

Reply:
getAllBy and querAllBy are not checked by the rule.

Reply:
it should be possible though right? at least for queryAllBy

Reply:
Yes, it's possible. I'll give it a try later and apply typos suggestions.

Reply:
cool and thanks again!

Reply:
@Belco90 do you think the first case i mentioned is possible as well?

i.e. 
```jsx
const button = screen.queryByText("button");
expect(button).toBeInTheDocument()
```

Reply:
Yes, still on it. I’ll request your review again when added